examples/javascript/create-preauth-req.js

```diff
@@ -0,0 +1,93 @@
+ *
+ * @param args Arguments to provide to the example. The following arguments are expected:
+ * <ul>
+ * <li>The first argument is the absloute directory path to read files from.</li>
+ * <li>The second argument is the name of an existing bucket name to generate the url.</li>
+ * </ul>
+ */
+
+const os = require('oci-objectstorage');
+const common = require('oci-common');
+const { readdir } = require('fs');
+
+const configurationFilePath = '~/.oci/config';
+const configProfile = 'DEFAULT';
+
+const provider = new common.ConfigFileAuthenticationDetailsProvider(
+  configurationFilePath,
+  configProfile
+);
+
+const args = process.argv.slice(2);
+if (args.length !== 2) {
+  console.error(
+    'Unexpected number of arguments received. Consult the script header comments for expected arguments'
+  );
+  process.exit(-1);
+}
+
+const directoryPath = args[0];
+const bucketName = args[1];
+const hostName = 'objectstorage'
+
+const client = new os.ObjectStorageClient({
+    authenticationDetailsProvider: provider
+});
+client.region = common.Region.US_PHOENIX_1;
+
+(async () => {
+    try {
+        console.log('Getting the namespace...');
+        const request = {};
+        const response = await client.getNamespace(request);
+        const namespaceName = response.value;
+        
+        // Read files from the directory
+        readdir(directoryPath, (err, files) => {
+            if (err) return console.log('Unable to scan directory: ' + err);
+
+            files.forEach(async filename => {
+                // creating pre authentication request which generates the download url for the file
+                console.time(`Download Time ${filename}`);
+
+                // set expiry date for the download url.
+                const today = new Date();
+                const neverExpires = new Date(today);
+                neverExpires.setDate(neverExpires.getDate() + 25);
+
+                // use date for generating a random unique id which can be used as a par id
+                const parUniqueId = Date.now();
+                const createPreauthenticatedRequestDetails = {
+                    name: parUniqueId.toString(),
+                    objectName: filename,
+                    accessType: os.models.CreatePreauthenticatedRequestDetails.AccessType.ObjectRead,
+                    timeExpires: neverExpires
+                };
+                const createPreauthenticatedRequest = {
+                    bucketName: bucketName,
+                    namespaceName: namespaceName,
+                    createPreauthenticatedRequestDetails: createPreauthenticatedRequestDetails
+                };
+                // create pre authenticated request to generate the url
+                const resp = await client.createPreauthenticatedRequest(createPreauthenticatedRequest);
+                const baseUrl = 'https://' + hostName + '.' + common.Region.US_PHOENIX_1.regionId + '.oraclecloud.com';
```

Comment:
Can the value ".oraclecloud.com" be replaced by common.Realm.OC1.secondLevelDomain here and also hostName variable with serviceName?